### PR TITLE
chore: Refactor reporter tests

### PR DESCRIPTION
**Pull request by chrisbreiding**
- Closes [TR-492](https://cypress-io.atlassian.net/browse/TR-492)

### User facing changelog

N/A - internal only

### Additional details

- Converts any `.js` specs/utilities to `.ts`
- Converts all enzyme `*.spec.ts` specs to cypress tests and adds percy snapshots

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? 
- N/A Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- N/A Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- N/A Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by chrisbreiding** [comment]
I don't know why the new percy snapshots are lacking CSS. I tried increasing the network-idle-timeout, which [had worked previously](https://github.com/cypress-io/cypress/pull/8469), but it doesn't seem to work.

**Review by jennifer-shehane — DISMISSED** [review]
I'd probably prefer a smaller PR in the future. This was a LOT to review and try to get straight since all the files changed. 

This looks good if we can figure out the CSS issue in Percy. 

I made some other comments that you can address if necessary.

File: packages/reporter/cypress/integration/test_errors_spec.js [hunk 1]
```diff
@@ -1,442 +0,0 @@
...
-
-    itHandlesFileOpening('.test-err-code-frame a', {
-      file: '/me/dev/my/app.js',
-      line: 2,
-      column: 7,
-    })
-  })
-})
-
-describe('test error without file info', function () {
```
[the remark this answers is not in the document]
Comment (chrisbreiding) [reply packages/reporter/cypress/integration/test_errors_spec.js:349]: They're covered by other tests and don't require separate fixtures/setup. This one ('test error without file info') is covered by [this test](https://github.com/cypress-io/cypress/blob/TR-492-refactor-reporter-tests/packages/reporter/cypress/integration/test_errors_spec.ts#L145) and the 'test error with good file info' one is covered by numerous tests in `test_errors_spec.ts` as well as in the isolated runner tests.

**Comment by chrisbreiding** [comment]
Was able to get the percy snapshots working properly 🙃 Needed to run the reporter tests with their own server and not the default Cypress server. I think it's because otherwise [percy thinks the CSS is being served by a different asset host](https://docs.percy.io/docs/capturing-multiple-asset-hosts).

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**